mesh/src/vertex/VertexMesh.cpp

```diff
@@ -1422,7 +1424,10 @@ unsigned VertexMesh<ELEMENT_DIM, SPACE_DIM>::GetLocalIndexForElementEdgeClosestT
 template <unsigned ELEMENT_DIM, unsigned SPACE_DIM>
 c_vector<double, 3> VertexMesh<ELEMENT_DIM, SPACE_DIM>::CalculateMomentsOfElement(unsigned index)
 {
-    assert(SPACE_DIM == 2); // LCOV_EXCL_LINE - code will be removed at compile time
+    if (SPACE_DIM != 2)
+    {
+        EXCEPTION("This function is only valid in 2D"); // LCOV_EXCL_LINE
```

Comment:
All `EXCEPTION`s should be covered as a rule, with a test that they throw the right exception, ideally?

Reply:
@mirams there were quite a few places that used to be:
```
assert(SPACE_DIM == 2); // LCOV_EXCL_LINE
```
that were changed to exceptions to satisfy the compiler about c_vector access.  So we're not ignoring anything more than we used to, it's just it was decided to move from assert -> exception.